thermotools/thermopack
# DFT interfaces

- Corrected typos in PC-SAFT memo
- Moved PC-SAFT parameters to json files
- Added script for generating pc_saft_datadb.f90
- Added methods to get saft dispersion contribution
- Added de Broglie wavelength method
- Added segment number, sigma and epsilon to pcsaft class
- Integrated pcsaft.py with pyUtils/datadb code
- Added option to calculate only ideal or residual pressure with pressure_tv method
- Minor fixes
- Removed pycharm files
- Reverted platform_specifics.py to original state
- Added logical parameters to play with contributions to PC-SAFT model
- Added methods for pure fluid Tc and Pc
- Added print of SAFT model parameters
- Made BMCSL hard-sphere module
- Added original PC-SAFT model
- Tested PC-SAFT and sPC-SAFT
- Updated code references in Makefile
- Updated PC-SAFT tests
- Added lng_ii interface for PC-SAFT
- Added Yu and Wu gij for FMT
- PC-SAFT associattion volume should be based on dij
- sPC-SAFT code does not use pi over 6 term
- Added interface to set HS-model
- Added option to use additional HS term with exact_binary_dhs enabled
- Added initial state class
- Added utility functions
- Initial version of mu-T solver
- Added potential function to saft interface
- Added notebook for hydrogen
- Added LJ potential for PC-SAFT
- Added interface to pure fluid envelope mapper
- Added interface to set de-Boer parameter
- Added notebooks for SAFT-VRQ Mie using additive reference
- Added missing mu-T solver
- Added method to extrapolate chemical potential for curved surfaces
- Bug fixes for state
- Added temperature differential to hard-sphere diameter method
- Always calculate d_T for python interface
- Corrected mu-T solver interfaces
- Corrected extrapolate_mu_in_inverse_radius
- Added Laplacew solver
- Added entropy methods to thermopack_state.py
- Updated notebooks
- Added soft-repulsion interface
- Added python interface to method for calculating binary triple point
- Added option to initialize SAFT-VRQ Mie with additive reference
- Added method for mapping pure fluid saturation curve
- Added more properties to state
- First iteration on pets functional
- Enabled and debugged PeTS interfaces
- Added memo about truncation corrections
- Activated truncation correction to the SAFT-VR Mie eos
- Added method to calculate reduced molar hard-sphere contribution to saft models
- Added print functionallity for equilibrium class
- Added LJs potential method
- Added method to test FMT compatibillity. Added hard-sphere delta_ij calculation.
- Added interface to RDF at contact functional
- Added energy-densities
- Two component fixes
- Chaced temperature and hard-sphere diameter for SAFT-VR Mie
- Added alpha interface for regular PC-SAFT
- Reduce CPU time consumption by looping only upper left part of dispersion-sums
- Added methods for mapping in meta-stable region
- Updated deuterium plots
- Corrected meta_curve code
- Added plots for for meta states
- Saving pets plots
- Added mu-T solver to thermopack state
- Corrected axis labels for deuterium plots
- Added FMT code
- Use capital first letter in python class
- Corrected FMT interface
- Added state example
- Latets plots
- Corrected test cases
- Enabled pytest case
- Staret adding association for FMT
- Debugged hyperdual LU solver
- Corrected temperature cache for saftvrmie
- Added inteface to association functional
- Added unittest for DFT
- my personal c_len_type
- Updated licence
- Added hyperdual numbers module
- Implemented HVWS cubic mixing rule, consistency checked
- Added kwargs for one-liner initialization of eos objects
- Modified pyExamples to use oneliner initialization
- Remove whitespace and unused code
- HVWS now uses kij and lij from underlying cubic EoS
- Add get and set routines for cubic covolume interaction parameter lij
- fixed indentation error
- added routine to set cubic lij
- Added option to set cpa formulation to simple
- Check if model%eos is allocated before accessing
- Added interface to covolume and energy constant for cubic eos
- Removed my personal c_len_type
- Modefied inport section of python scripts and started adding minimum_temperature as input to constructor and init
- Added docstrings to __init__ methods in pyctp
- Added temperature_minimum for quantum cubic init, and corrected wca init.
- Nested functions not ISO C standard
- Added LJ and LJTS EOS
- Tweaking use of LJ model
- Adapted code for intel compiler and windows build
- Assume GCC > 7
- Added wsparam get/set methods
- Initialize lambda in envelopePlot
- Added MSVS project for executable. Modefied MSVS floating point settings to get similar bahaviour as with linux builds
- Generate all plots at end of scripts
- Fixed typo in exports and regenerated libthermopack_export.* files
- Typos in CONTRIBUTING.md
- Update README.md with thermotools repository location
- Renamed SINTEF/thermopack to thermotools/thermopack
- Added alternative Makefiles to compile for Mac with arm64 architecture
- Added readme instructions for building for mac with arm64
- Generate pure fluid phase diagrams with solid
- Reverse order of liquid phase
- Added density temperature plot
- Corrected some issues with envelope plotting with solids
- Added Temperature density envelope for mixture
- Added solid-gas curve for nc=2
- Added pyproject.toml file
- Setup working for both windows and manylinux wheel generation
- Update README.md
- Tweaked configuration for wheels
- Renaming python interface to thermopack
- Deleted pyCharm folders
- Ignoring .idea
- Updated gitignore to include wheels and build files. Also updated path to platform_specifics.py
- Set reference entropy/ethalpy on init
- Updated links and credited NTNU
- Added reference entropy and enthalpy for N2 and NH3
- Added changelog to document break of backward compatibillity with python package name
- Bumped version number to 2.1.0
- Updated changelog
- Added multiparameter_lj source files to MSVStudio projects
- Adding missing exports
- Added readme to pypi long_description
- Improve critical point solver robustness when tuning sigma/eps
- Corrected integration limit for cavity term
- Added set and get methods for tmax used in numerical solvers
- Added min/max temp/press to docstring for get/set methods
- Correcting interfaces
- Added pressure as member in Flashresult
- Accept that tolerence is not met in entropy-pressure solver
- Made improvements to code mapping spinodal
- Corrected test for if temperature is out of range
- Added LNG RPT example
- Updated arm64 makefile according to changes in x86 makefile
- Removed code not belonging in Thermopack
- Added method to get all critical parameters
- Added doc-tag to new methods in thermo.py
- Corrected dT/dP stepping for envelope mapping
- Corrected notebooks
- Added scaling of default step size for envelope-plotting
- Removed dummy variables
